android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt

```diff
@@ -270,7 +286,73 @@ class ExecuteSendFromOrchard {
         System.out.println("\nBalance post-send:")
         System.out.println(balanceJson)
         val balancePostSend: Balance = mapper.readValue(balanceJson)
+        assertThat(balancePreSend.verified_orchard_balance).isEqualTo(670000)
+        assertThat(balancePreSend.spendable_orchard_balance).isEqualTo(670000)
```

Comment:
have you confirmed this assertion? "spendable" means it has a confirmation right? AFAIK no blocks have been generated (or mined) since the send so i assume it is not confirmed...

if this isnt the case it would be interesting to know whats happening here

Reply:
well, I don't know yet honestly... I'm waiting for the first run... but thanks, because it's interesting what you are saying...

Reply:
balance:
09-28 15:21:31.405  9564  9589 I System.out: Balance post-send:
09-28 15:21:31.405  9564  9589 I System.out: {
09-28 15:21:31.405  9564  9589 I System.out:   "sapling_balance": 100000,
09-28 15:21:31.405  9564  9589 I System.out:   "verified_sapling_balance": 100000,
09-28 15:21:31.405  9564  9589 I System.out:   "spendable_sapling_balance": 100000,
09-28 15:21:31.405  9564  9589 I System.out:   "unverified_sapling_balance": 0,
09-28 15:21:31.406  9564  9589 I System.out:   "orchard_balance": 670000,
09-28 15:21:31.406  9564  9589 I System.out:   "verified_orchard_balance": 0,
09-28 15:21:31.406  9564  9589 I System.out:   "spendable_orchard_balance": 0,
09-28 15:21:31.406  9564  9589 I System.out:   "unverified_orchard_balance": 670000,
09-28 15:21:31.406  9564  9589 I System.out:   "transparent_balance": 100000
09-28 15:21:31.406  9564  9589 I System.out: }

Reply:
I'm checking only the `orchard_balance` right after the `send` process... If we want to check `verified` or even `spendable` we need to wait for a while, and the test is already too long...

Reply:
i see that there are verified sapling funds after the send and the test proves this? also have you tested that the orchard verified and spendable balance are in fact equal to 670000 after some time?

the reason i ask this is because i am curious to know if verified/spendable balances can change without additional blocks being generated/mined. i see three possibilities:
1. how we update balances on sending doesn't require confirmations, only receiving does (this would be strange).
2. zcashd is somehow configured to generate blocks after sends (i find this case highly unlikely)
3. i am completely missing something or misunderstanding everything

i think 3. is most likely haha lets see what everyone thinks....

Reply:
resolving conversation so i can merge ... lets continue this conversation in the chat!